internal/handler/controller/domain.go

```diff
@@ -5,40 +5,84 @@ import (
 	"net/http"
 
 	"github.com/go-chi/chi/v5"
+	"github.com/oursky/pageship/internal/api"
 	"github.com/oursky/pageship/internal/db"
 	"github.com/oursky/pageship/internal/models"
 	"go.uber.org/zap"
 )
 
-type apiDomain struct {
-	*models.Domain
+func (c *Controller) makeAPIDomain(domain *models.Domain, dominVerification *models.DomainVerification) *api.APIDomain {
+	return &api.APIDomain{Domain: domain, DomainVerification: dominVerification}
 }
 
-func (c *Controller) makeAPIDomain(domain *models.Domain) *apiDomain {
-	return &apiDomain{Domain: domain}
+type apiDomainVerification struct {
+	*models.DomainVerification
+}
+
+func (c *Controller) makeAPIDomainVerification(domainVerification *models.DomainVerification) *apiDomainVerification {
+	return &apiDomainVerification{DomainVerification: domainVerification}
+}
+
+type DomainInfo struct {
+	domain       *models.Domain
+	verification *models.DomainVerification
 }
 
 func (c *Controller) handleDomainList(w http.ResponseWriter, r *http.Request) {
 	app := get[*models.App](r)
 
 	respond(w, func() (any, error) {
-		var domains []*models.Domain
+		var domains []DomainInfo
 		for _, dconf := range app.Config.Domains {
 			domain, err := c.DB.GetDomainByName(r.Context(), dconf.Domain)
-			if errors.Is(err, models.ErrDomainNotFound) {
-				continue
-			} else if err != nil {
+			if err != nil && !errors.Is(err, models.ErrDomainNotFound) {
 				return nil, err
 			}
-			domains = append(domains, domain)
+			domainVerification, err := c.DB.GetDomainVerificationByName(r.Context(), dconf.Domain)
+			if err != nil && !errors.Is(err, models.ErrDomainNotFound) {
+				return nil, err
+			}
+			if domain != nil || domainVerification != nil {
+				domains = append(domains, DomainInfo{domain: domain, verification: domainVerification})
+			}
 		}
 
-		return mapModels(domains, func(d *models.Domain) *apiDomain {
-			return c.makeAPIDomain(d)
+		return mapModels(domains, func(d DomainInfo) *api.APIDomain {
+			return c.makeAPIDomain(d.domain, d.verification)
 		}), nil
 	})
 }
 
+func (c *Controller) handleDomainActivate(w http.ResponseWriter, r *http.Request) {
+	app := get[*models.App](r)
+	domainName := chi.URLParam(r, "domain-name")
+
+	config, ok := app.Config.ResolveDomain(domainName)
+	if !ok {
+		respond(w, func() (any, error) { return nil, models.ErrUndefinedDomain })
+	}
+	if c.Config.DomainVerification {
```

Comment:
I would not expect the activate API would also create the domain; the API flow should be like this:
1. User update app config in CLI
2. Call update config API
3. User activates domain in CLI
4. Call create domain API
5. If domain verification is not enabled: API create domain association.
6. If domain verification is enabled: API create domain verification request.
7. User see the TXT record in CLI
8. User update DNS with the TXT record
9. User activates domain in CLI
10. Call create domain API
11. Domain verification request present: mark the domain verification request as pending.